awx/main/models/ad_hoc_commands.py

```diff
@@ -182,7 +187,10 @@ def get_passwords_needed_to_start(self):
     def task_impact(self):
         # NOTE: We sorta have to assume the host count matches and that forks default to 5
         from awx.main.models.inventory import Host
-        count_hosts = Host.objects.filter( enabled=True, inventory__ad_hoc_commands__pk=self.pk).count()
+        if self.host_impact is None:
+            self.host_impact = Host.objects.filter( enabled=True, inventory__ad_hoc_commands__pk=self.pk).count()
+            self.save(update_fields=['host_impact'])
+        count_hosts = self.host_impact
```

Comment:
Let me just throw out an alternative location to put this.

This calculation could be done right before the same performed in `create_unified_job`.

https://github.com/ansible/awx/blob/53522e4b781d85b9372f81b62872cc4dd2cbd549/awx/main/models/unified_jobs.py#L384

For a single job, this should generally be a fairly fast calculation (although I we have had slow response times for the POST creating a job before). Also, doing it this way would avoid an additional save, by piggybacking on the job creation save.

The thing that bothers me about this implementation is that we wind up writing fields when doing a GET, and it makes it generally more difficult to predict what queries a particular action will result in. Everything's a tradeoff, so I understand if you still want to do it this way. I know `create_unified_job` is extremely bloated.

Reply:
Yep, I agree that this makes more sense when the job is created.